### PR TITLE
Move ActiveRecord::Type to ActiveModel

### PR DESCRIPTION
**Pull request by kirs**
The first step of bringing typecasting to ActiveModel.
The next step would be to move typecasted attribute assignment and make that API public.

/cc @sgrif @rafaelfranca

### CODE REVIEW

File: activemodel/lib/active_model/type/registry.rb [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+
+      attr_reader :registrations
+      
+      private
+      
+      def registration_klass
+        Registration
+      end
+      
+      def recoration_registration_klass
```
Comment (kaspth) [inline activemodel/lib/active_model/type/registry.rb]: Should be decoration, right?

**Comment by dmitry** [comment]
On this stage it's good to take into account multiparameters movement: https://github.com/rails/rails/pull/19709

**Comment by kirs** [comment]
@dmitry that's my PR actually

**Comment by sgrif** [comment]
Thanks kirs, this is going to take some time to review (I'm also sick and keeping away from code right now). Is it OK with you if I just fix any stylistic nitpicks if that's all I have?

**Comment by kirs** [comment]
@sgrif sure, thanks! I tried to make the changeset as small as possible as you suggested back in April.
Get better! :star2:

**Comment by kirs** [comment]
@sgrif how are you? Please tell if there is anything that I can do.

**Comment by sgrif** [comment]
Just got back from out of town. I have this in an open tab and will be
taking care of it this week.

**Pull request closed without merging**